### ISSUE TITLE
This plugin cannot select multiple dates on iOS

### ISSUE THREAD

**Issue by cnqiuxue** [NativeScript/nativescript-ui-feedback#728]
This plugin cannot select multiple dates on iOS

![image](https://user-images.githubusercontent.com/26665435/42566464-48a8c30c-8539-11e8-9e44-044961f08a1e.png)
```
<StackLayout class="page">
        <StackLayout class="calendar" tkExampleTitle tkToggleNavButton>
                <RadCalendar #myRalendar #myCalendar [selectedDates]="selectedDates" (dateSelected)="onDateSelected($event)" 
                selectionMode="Range" [eventSource]="eventSource" (Swipe)="setEventSource($event)" (loaded)="calendarloaded($event)" 
                (viewModeChanged)="onViewModeChangedEvent($event)" [selectionMode]="selectionMode"></RadCalendar>
        </StackLayout>
        <ListView class="list-group" [items]="myItems">
                <ng-template let-country="item">
                        <FlexboxLayout class="list-group-item">
                                <FlexboxLayout width="50%" flexDirection="column" alignSelf="center">
                                        <Label class="title" [text]="country.title"></Label>
                                </FlexboxLayout>
                        </FlexboxLayout>
                </ng-template>
        </ListView>
</StackLayout>
```
```
selectedDates: Array<string> = [];
const yiCalendar = this._ramadanEService.getCalendarByYiCalendar("/9/1");
        if (yiCalendar) {
            const len = this._ramadanEService.monthIs30Day("/9/30") ? 30 : 29;
            let addDay = moment(yiCalendar);
            for (let i = 1; i <= len; i++) {
                this.selectedDates.push(addDay.format("YYYY/M/D"));
                addDay = moment(addDay).add(1, "d");
            }
        }
```

**Comment by tsonevn** [comment]
HI @cnqiuxue,
I tested the multiple date selection with our [sample project](https://github.com/telerik/nativescript-ui-samples-angular/tree/master/calendar/app/calendar/selection-modes), however, I was unable to recreate an issue with this functionality. 
It will help if you provide a sample project, where the problem can be reproduced.

**Comment by cnqiuxue** [comment]
@tsonevn 
The multi-date selection I want to implement is shown on Android as follows
![screenshot_20180718-101634](https://user-images.githubusercontent.com/26665435/42855648-1746c8f0-8a74-11e8-99a2-10d47026ce56.jpg)
![screenshot_20180718-101632](https://user-images.githubusercontent.com/26665435/42855649-1937be94-8a74-11e8-9bb7-7db3a0a7ef60.jpg)
But ios can't implement such a function, the code is the same
https://user-images.githubusercontent.com/26665435/42566464-48a8c30c-8539-11e8-9e44-044961f08a1e.png

**Comment by tsonevn** [comment]
Hi @cnqiuxue,
In iOS, you should select each day while using multiple selection mode. For your convenience, I am attaching GIF file.
![screencast 2018-07-18 at 11 22 32 am](https://user-images.githubusercontent.com/17448734/42875492-90f2e568-8a8c-11e8-8fe7-39aaa0f0bd66.gif)

**Comment by cnqiuxue** [comment]
hi@tsonevn 
I wrote it with the SelectedDates attribute, unfortunately, iOS didn't work.
Android's normal.

**Comment by zbranzov** [comment]
Programmatically setting the selected date does not work on iOS  "version": "3.10.0"
```
onLoaded(args) {
        console.log("=====================");
        var today = new Date();
        var tomorrow = new Date();
        tomorrow.setDate(today.getDate()+1);
        this._calendar.calendar.selectedDate = tomorrow;
        this._calendar.nativeElement.reload();
    }
```

**Comment by tgpetrov** [comment]
Hi @cnqiuxue 
I believe you are using Range selection mode. If that is the case, you can set selection programmatically through selectedDateRange and not through selectedDates. We are going to add a warning when an attribute that is not compatible with the current selection mode is used in order to avoid confusion.
@zbranzov 
You have found another bug. According to [the docs](https://docs.nativescript.org/ns-ui-api-reference/classes/radcalendar#reload), calling reload should clear the selection, but it seems to not work on android, so the selection remains.

**Comment by zbranzov** [comment]
Released in nativescript-ui-calendar@4.1.0
